packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailComposerModule.kt

```diff
@@ -72,7 +81,53 @@ class MailComposerModule : Module() {
     }
   }
 
+  // Function to get available mail clients
+  private fun getAvailableMailClients(): List<MailClient> {
+    val emailIntent = Intent(Intent.ACTION_SENDTO).apply {
+      data = Uri.parse(MAILTO_URI)
+    }
+    val pm = context.packageManager
+    val resolveInfoList = pm.queryIntentActivities(emailIntent, 0)
+
+    return resolveInfoList.map { resolveInfo ->
+      val packageName = resolveInfo.activityInfo.packageName
+      val appInfo = pm.getApplicationInfo(packageName, 0)
+      val appName = pm.getApplicationLabel(appInfo).toString()
+
+      // Get the app icon as a base64-encoded string
+      val iconDrawable = resolveInfo.loadIcon(pm)
+      val iconBase64 = iconDrawable?.let { drawable ->
```

Comment:
@lukmccall may have some thoughts about exposing the icon. It might be a bit expensive for a synchronous function, especially when the app icon is big.

Reply:
What do you think about exposing the app icon API through the intent launcher? Currently, the icon is converted to a bitmap, which can take some time. Additionally, not everyone might need that functionality, and it seems to be unrelated to the mail composer itself. It would be better to expose it through a different module, and the mail composer could only return the label and the package name without the icon.

Reply:
I agree. I’ve removed the app icon functionality from the `expo-mail-composer` and moved it into the `expo-intent-launcher` module. It’s now implemented as an asynchronous function.